classes/web_processor.php

```diff
@@ -73,14 +80,17 @@ public function get_min_duration(): float {
      * @throws \dml_exception
      */
     public function process(manager $manager, bool $isfinal): void {
-        $log = $manager->get_profiler()->getLog();
+        $log = $manager->get_profiler()->flush();
+        foreach ($log as $entry) {
+            $this->sampleset->add_sample($entry);
+        }
```

Comment:
Looking at this and the tests, I think it would be good to have a `add_many_samples` function or similar, to do handle this loop consistently all the time. This would also reduce the logic done in general (since flush and samples need to always be added like this) and also would simplify the handling in tests.